mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/resourceprovider/SimpleFileResourceClusterStorageProvider.java

```diff
@@ -212,7 +213,7 @@ public CompletionStage<ResourceClusterScaleRulesWritable> registerResourceCluste
             Source
                 .single(ruleSpec)
                 .mapAsync(1, this::updateClusterScaleRules)
-                .mapAsync(1, rc -> getResourceClusterScaleRules(ruleSpec.getClusterId()))
+                .mapAsync(1, rc -> getResourceClusterScaleRules(ClusterID.of(ruleSpec.getClusterId())))
```

Comment:
nit: maybe as part of a followup PR, can we ensure ResourceClusterScaleRulesWritable::getClusterId is also of type ClusterID?

Reply:
I was thinking about this one and the extra wrapping for the file to be written as the primary key in the datastore might get unexpected problems with the same prefix and extra format chars, which is why I kept the string format instead. The logic level should be fine with strong type and we can isolate for *writable.

Reply:
Updated in current PR.